File: VDFEditor/src/main/java/de/treichels/hott/vdfeditor/ui/Text2SpeechTask.java
package de.treichels.hott.vdfeditor.ui;

import java.io.File;

import gde.model.voice.VoiceRSS;
import gde.model.voice.VoiceRssLanguage;
import javafx.concurrent.Task;

final class Text2SpeechTask extends Task<File> {
    private final VoiceRSS voiceRSS = new VoiceRSS();

    @Override
    protected File call() throws Exception {
        return voiceRSS.getFile();
    }

    public VoiceRssLanguage getLanguage() {
        return voiceRSS.getLanguage();
    }

    public String getText() {
        return voiceRSS.getText();
    }

    void setLanguage(final VoiceRssLanguage language) {
        voiceRSS.setLanguage(language);
    }

    public void setText(final String text) {
        voiceRSS.setText(text);
    }
}

File: VDFEditor/src/main/java/de/treichels/hott/vdfeditor/ui/tts/Text2SpeechTask.kt
package de.treichels.hott.vdfeditor.ui.tts

import gde.model.voice.VoiceRSS
import gde.model.voice.VoiceRssLanguage
import javafx.concurrent.Task
import java.io.File

internal class Text2SpeechTask : Task<File>() {
    private val voiceRSS = VoiceRSS()

    var language: VoiceRssLanguage
        get() = voiceRSS.language
        set(language) {
            voiceRSS.language = language
        }

    var text: String
        get() = voiceRSS.text
        set(text) {
            voiceRSS.text = text
        }

    var volume = 1.0

    var speed: Int
        get() = voiceRSS.speed
        set(speed) {
            voiceRSS.speed = speed
        }

    override fun call(): File = voiceRSS.file
}
